runtime/js/99_main.js

```diff
@@ -89,6 +93,36 @@ ObjectDefineProperties(Symbol, {
 let windowIsClosing = false;
 let globalThis_;
 
+const ALREADY_WARNED_DEPRECATED = new SafeSet();
+
+function warnOnDeprecatedApi(apiName, stack) {
+  const stackLines = StringPrototypeSplit(stack, "\n");
+  ArrayPrototypeShift(stackLines);
```

Comment:
Curious why we shift one item here

Reply:
This is to remove the `Error\n` at the start of the output from `new Error().stack`.

Reply:
I wonder if that's worth the cost of `split`, `shift`, and `join`

Reply:
I suggest `stack.slice(6)` with a comment.

Reply:
I also suggest we should do that `slice` at the line of `"%cThis API was called from:\n" + stackString + "\n",` to minimize the overhead of this util

Reply:
I will change the logic to first check if such stack traces was produced before manipulating the stack.